src/category_theory/eq_to_hom.lean

```diff
@@ -5,6 +5,26 @@ Authors: Reid Barton, Scott Morrison
 -/
 import category_theory.opposites
 
+/-!
+# Morphisms from equations between objects.
+
+When working categorically, sometimes one encounters an equation `h : X = Y` between objects.
+
+Your initial aversion to this is natural and appropriate:
+you're in for some trouble, and if there is another way to approach the problem that won't
+rely on this equality, it may be worth pursuing.
+
+You have two options:
+1. Use the equality `h` as one normally would in Lean (e.g. using `rw` and `subst`).
+   This may immediately cause difficulties, because in category theory everything is dependently
+   typed, and equations between objects quickly lead to nasty goals with `eq.rec`.
+2. Promote `h` to a morphism using `eq_to_hom h : X ⟶ Y`, or `eq_to_iso h : X ≅ Y`.
```

Comment:
I think the precise point is that it rather **demotes** the equality, right?

Reply:
Stand on your head? I'm not sure what you mean by "demote" rather than "promote".

Reply:
Promoting is buffing up, adding more infos or, in a more mathliby context, bundling more an existing how for example. Demoting is weakening, debuffing, losing infos. In that case, an isomorphism or a morphism is weaker than an equality.